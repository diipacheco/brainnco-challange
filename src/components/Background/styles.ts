import styled from 'styled-components';

import { IBackgroundProps } from './index';

export const backgroundColors = {
  'MEGA-SENA': 'var(--green-light)',
  QUINA: 'var(--purple-dark)',
  LOTOFÁCIL: 'var(--purple-light)',
  LOTOMANIA: 'var(--orange)',
  TIMEMANIA: 'var(--green-dark)',
  'DIA DE SORTE': 'var(--yellow-ocre)',
};

export const Container = styled.div<IBackgroundProps>`
  width: 100%;
  height: 390px;

  display: flex;
  flex-direction: column;
  align-items: center;
  position: relative;

  background-color: ${props => {
    switch (props.selectedLottery) {
      case 'MEGA-SENA':
        return backgroundColors['MEGA-SENA'];
      case 'QUINA':
        return backgroundColors.QUINA;
      case 'LOTOFÁCIL':
        return backgroundColors.LOTOFÁCIL;
      case 'LOTOMANIA':
        return backgroundColors.LOTOMANIA;
      case 'TIMEMANIA':
        return backgroundColors.TIMEMANIA;
      case 'DIA DE SORTE':
        return backgroundColors['DIA DE SORTE'];
      default:
        return backgroundColors['MEGA-SENA'];
    }
  }};

  @media screen and (min-width: 700px) {
    height: 100vh;
    max-width: 550px;

    &::after {
      content: '';
      width: 40px;
      height: 100%;
      background-color: var(--background);
      position: absolute;
      border-radius: 62%;
      right: -19px;
    }
  }

  @media screen and (max-width: 700px) {
    &::after {
      content: '';
      width: 100%;
      height: 52px;
      background-color: var(--background);
      position: absolute;
      border-radius: 40%;
      top: 92%;
    }
  }
`;
