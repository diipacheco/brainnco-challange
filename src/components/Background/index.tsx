import React from 'react';

import { Container } from './styles';

export interface IBackgroundProps {
  selectedLottery: string | undefined;
}

const Background: React.FC<IBackgroundProps> = ({
  selectedLottery,
  children,
}) => {
  return (
    <>
      <Container selectedLottery={selectedLottery}>{children}</Container>
    </>
  );
};

export default Background;
