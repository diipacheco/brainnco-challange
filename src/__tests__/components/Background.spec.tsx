import { render } from '@testing-library/react';

import Background from '../../components/Background';

describe('Background', () => {
  it('should match snapshots', () => {
    const wrapper = render(<Background selectedLottery="megaSena" />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to render with the specific background color from the selected lottery', () => {
    const { container } = render(<Background selectedLottery="megaSena" />);

    expect(container.firstChild).toHaveStyleRule(
      'background-color',
      'var(--green-light)',
    );
  });
});
