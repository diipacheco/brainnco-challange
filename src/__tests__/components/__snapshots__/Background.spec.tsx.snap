// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background should match snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  width: 100%;
  height: 390px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  background-color: var(--green-light);
}

@media screen and (min-width:700px) {
  .c0 {
    height: 100vh;
    max-width: 550px;
  }

  .c0::after {
    content: '';
    width: 40px;
    height: 100%;
    background-color: var(--background);
    position: absolute;
    border-radius: 62%;
    right: -19px;
  }
}

@media screen and (max-width:700px) {
  .c0::after {
    content: '';
    width: 100%;
    height: 52px;
    background-color: var(--background);
    position: absolute;
    border-radius: 40%;
    top: 92%;
  }
}

<body>
    <div>
      <div
        class="c0"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM hxsYFM"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
